src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final Object[] EMPTY_SIZE = {};
+    private int size = 0;
+    private Object[] values = new Object[DEFAULT_SIZE];
```

Comment:
It will be better to store parametrized value in array, to avoid casting every time you want to interact with array values